### PR TITLE
Add initial basic CI

### PR DESCRIPTION
**Pull request by henrygoodman**
Just using the `package.json` scripts `next build` (which also runs linting I believe) to double check code is good before merging onto main.

### CODE REVIEW

**Comment by henrygoodman** [comment]
@CDo246 normally the CI would run in a PR to `main` like this, I'm unsure if a setting needs to be enabled in order for it to work, or if it's not working since there is no actions within the `main` branch yet (maybe see if there are any settings?)

I marked this as a Work In Progress PR just so I could do some testing on the CI and finetune it if not working, however cannot do much testing in this state. 

If you're happy to go along with this, I think we can merge this into main, and then we should see if any future PR's trigger it. If they do, we can test it and fix it up if not working, and if the CI doesn't trigger, we will look more into it

EDIT: Disregard above, there was a typo in the folder name. It usually is something simple like that

(Also the other changes in this PR are linting errors that have been fixed.)

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Node.js CI
+
+on:
+  push:
```
Comment (CDo246) [inline .github/workflows/node.js.yml]: Might not need to check code twice, both on pull requests/push.

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
```
Comment (CDo246) [inline .github/workflows/node.js.yml:11]: Would probably also be a good idea to have a step for checking formatting via prettier

Comment (henrygoodman) [reply .github/workflows/node.js.yml:11]: Added `prettier:check` command in `package.json` that runs prettier. If this check fails, the dev will likely need to run `yarn prettier:format`. 

Majority of file changes in this PR are due to prettier format being run and applying the format changes.

File: .github/workflows/node.js.yml (hunk 1 above)
Comment (henrygoodman) [reply .github/workflows/node.js.yml]: Changed so now only running on PR

**Comment by henrygoodman** [comment]
Making a note in Disc for this, but going forward, devs should run the following before pushing changes to main (fine if not, the CI will just complain until its fixed)

```
yarn build (to test building/linting)
yarn prettier:format
yarn prettier:check
```

(If you merge this, create squash merge or create merge commit pls, a lot of junk commits in here)

**Comment by CDo246** [comment]
Would you be able to fix the build check error and resubmit? There are likely merge conflicts

**Comment by henrygoodman** [comment]
@CDo246 I've merged the state of main so no more conflicts, the CI will fail for this as there are some linting fails from the latest PR onto main. I will create another commit a bit later to fix those linting errors.

**Comment by henrygoodman** [comment]
This should now be good to merge, made some minor changes to fix linting as well as disabled some very semantic linting notifications.

**Review by CDo246 — APPROVED** [review]

**Pull request merged**